src/compile_error_kind.rs

```diff
@@ -98,6 +98,9 @@ pub(crate) enum CompileErrorKind<'src> {
     alias: &'src str,
     target: &'src str,
   },
+  UnknownAttribute {
+    attribute: &'src str,
```

Comment:
```suggestion
    attribute: Name<'src>,
```

Reply:
Other `CompileErrorKind`s like `UnknownFunction`, `UnknownSetting` all have `&'src str`. I thought there was nothing special about the attribute. Considering your `parse_attribute_name` suggestion, do you find anything wrong or ugly with the code below @casey?

```rs
...
UnknownAttribute {
  attribute: &'src str,
},
```

```rs
  /// Parse a recipe attribute name
  fn parse_attribute_name(&mut self) -> CompileResult<'src, Attribute> {
    let name = self.parse_name()?;
    Attribute::from_name(name).ok_or_else(|| {
      name.error(CompileErrorKind::UnknownAttribute {
        attribute: name.lexeme(),
      })
    })
  }
```

I think we should have `&'src str` here. `CompileError` returned by  `error()` method already has a token field. We should not have another `Token` or `Name` type field within the `CompileErrorKind`.

```rs
pub(crate) struct CompileError<'src> {
  pub(crate) token: Token<'src>,
  pub(crate) kind: Box<CompileErrorKind<'src>>,
}
```

Reply:
Great point, makes sense to me!